server/src/main/java/com/microsoft/java/bs/core/internal/utils/ConversionUtils.java

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+package com.microsoft.java.bs.core.internal.utils;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
+import com.microsoft.java.bs.gradle.model.impl.DefaultJavaExtension;
+
+/**
+ * Utils to convert objects between two different class loaders to avoid
+ * ClassCastException.
+ */
+public class ConversionUtils {
+
+  /**
+   * Convert an object to a DefaultJavaExtension.
+   */
+  public static DefaultJavaExtension toJavaExtension(Object object) {
```

Comment:
Same as in `Conversions` - why not do this copy in DefaultGradleSourceSet constructor like the other classes are copied?